src/VirtualDevice.ts

```diff
@@ -138,12 +138,15 @@ export class VirtualDevice {
         });
     }
 
-    public resetSession(locale?: string): Promise<IVirtualDeviceResult> {
-        let message = "quit";
-        if (locale && locale === "de-DE") {
-            message = "stopp";
+    public waitForSessionToEnd() {
+        const ms: number = process.env.SESSION_IDLE_MS
+        ? parseInt(process.env.SESSION_IDLE_MS, 10)
+        : 8000;
+        const start = new Date().getTime();
```

Comment:
you can combine setTimeout and promises for this one to avoid looping a lot,
public async waitForSessionToEnd() {
const ms: number = < ms logic >
return new Promise((resolve) => setTimeout(resolve, ms));
}